src/Plugins/ComplexCore/src/ComplexCore/Filters/Algorithms/ReplaceElementAttributesWithNeighborValues.cpp

```diff
@@ -0,0 +1,267 @@
+  IComparisonFunctor& operator=(const IComparisonFunctor&) = delete; // Copy Assignment Not Implemented
+  IComparisonFunctor& operator=(IComparisonFunctor&&) = delete;      // Move Assignment Not Implemented
+
+  virtual bool compare(T a, T b) const = 0;
+  virtual bool compare1(T a, T b) const = 0;
+  virtual bool compare2(T a, T b) const = 0;
+};
+
+template <typename T>
+class LessThanComparison : public IComparisonFunctor<T>
+{
+public:
+  LessThanComparison() = default;
+  virtual ~LessThanComparison() = default;
+
+  LessThanComparison(const LessThanComparison&) = delete;            // Copy Constructor Not Implemented
+  LessThanComparison(LessThanComparison&&) = delete;                 // Move Constructor Not Implemented
+  LessThanComparison& operator=(const LessThanComparison&) = delete; // Copy Assignment Not Implemented
+  LessThanComparison& operator=(LessThanComparison&&) = delete;      // Move Assignment Not Implemented
+
+  bool compare(T a, T b) const override
+  {
+    return a < b;
+  }
+  bool compare1(T a, T b) const override
+  {
+    return a >= b;
+  }
+  bool compare2(T a, T b) const override
+  {
+    return a > b;
+  }
+};
+
+template <typename T>
+class GreaterThanComparison : public IComparisonFunctor<T>
+{
+public:
+  GreaterThanComparison() = default;
+  ~GreaterThanComparison() override = default;
+  GreaterThanComparison(const GreaterThanComparison&) = delete;            // Copy Constructor Not Implemented
+  GreaterThanComparison(GreaterThanComparison&&) = delete;                 // Move Constructor Not Implemented
+  GreaterThanComparison& operator=(const GreaterThanComparison&) = delete; // Copy Assignment Not Implemented
+  GreaterThanComparison& operator=(GreaterThanComparison&&) = delete;      // Move Assignment Not Implemented
+
+  bool compare(T a, T b) const override
+  {
+    return a > b;
+  }
+  bool compare1(T a, T b) const override
+  {
+    return a <= b;
+  }
+  bool compare2(T a, T b) const override
+  {
+    return a < b;
+  }
+};
+
+struct ExecuteTemplate
+{
+
+  template <typename T>
+  void operator()(const DataStructure& dataStructure, const ImageGeom& imageGeom, IDataArray& inputIDataArray, int32 comparisonAlgorithm, float thresholdValue, bool loopUntilDone,
+                  const std::atomic_bool& shouldCancel, const IFilter::MessageHandler& messageHandler)
+  {
+    using DataArrayType = DataArray<T>;
+
+    const DataArrayType& inputArray = dynamic_cast<const DataArrayType&>(inputIDataArray);
+
+    size_t totalPoints = inputArray.getNumberOfTuples();
+
+    Vec3 udims = imageGeom.getDimensions();
+    int64_t dims[3] = {
+        static_cast<int64_t>(udims[0]),
+        static_cast<int64_t>(udims[1]),
+        static_cast<int64_t>(udims[2]),
+    };
+
+    bool good = true;
+    int64_t neighbor = 0;
+    int64_t column = 0, row = 0, plane = 0;
+
+    int64_t neighborPoints[6] = {0, 0, 0, 0, 0, 0};
```

Comment:
```suggestion
    int64_t neighborPoints[6] = {};
```
c++ initializes all elements to zero implicity when provided an empty constructor

Reply:
This was refactored